STEEL.md

```diff
@@ -0,0 +1,166 @@
+
+To install the `hx` executable, with steel as a plugin language. This also includes:
+
+The `steel` executable, the steel language server, the steel dylib installer, and the steel standard library.
+
+
+## Setting up configurations for helix
+
+Note, this API is entirely subjet to change, and I promise absolutely 0 backwards compatibility while this is in development.
+
+There are 2 important files you'll want, which should be auto generated during the installation process:
+
+* `~/.config/helix/helix.scm`
+* `~/.config/helix/init.scm`
+
+Note - these both live inside the same directory that helix sets up for runtime configurations.
+
+
+### `helix.scm`
+
+The `helix.scm` module will be loaded first before anything else, the runtime will `require` this module, and any functions exported will now be available
+to be used as typed commands. For example:
+
+
+```scheme
+# helix.scm
+(require "helix/editor.scm")
+(require (prefix-in helix. "helix/commands.scm"))
+(require (prefix-in helix.static. "helix/static.scm"))
+
+(provide shell git-add open-helix-scm open-init-scm)
+
+;;@doc
+;; Specialized shell implementation, where % is a wildcard for the current file
+(define (shell cx . args)
+  ;; Replace the % with the current file
+  (define expanded (map (lambda (x) (if (equal? x "%") (current-path cx) x)) args))
+  (apply helix.run-shell-command expanded))
+
+;;@doc
+;; Adds the current file to git	
+(define (git-add cx)
+  (shell cx "git" "add" "%"))
+
+(define (current-path)
+  (let* ([focus (editor-focus)]
+         [focus-doc-id (editor->doc-id focus)])
+    (editor-document->path focus-doc-id)))
+
+;;@doc
+;; Open the helix.scm file
+(define (open-helix-scm)
+  (helix.open (helix.static.get-helix-scm-path)))
+
+;;@doc
+;; Opens the init.scm file
+(define (open-init-scm)
+  (helix.open (helix.static.get-init-scm-path)))
+  
+	
+```
+
+Now, if you'd like to add the current file you're editing to git, simply type `:git-add` - you'll see the doc pop up with it since we've annotated the function
+with the `@doc` symbol. Hitting enter will execute the command.
+
+You can also conveniently open the `helix.scm` file by using the typed command `:open-helix-scm`.
+
+
+### `init.scm`
+
+The `init.scm` file is run at the top level, immediately after the `helix.scm` module is `require`d. The helix context is available here, so you can interact with the editor.
+
+The helix context is bound to the top level variable `*helix.cx*`.
+
+For example, if we wanted to select a random theme at startup:
+
+```scheme
+# init.scm
+
+(require-builtin steel/random as rand::)
+(require (prefix-in helix. "helix/commands.scm"))
+(require (prefix-in helix.static. "helix/static.scm"))
+
+(define rng (rand::thread-rng!))
+
+;; Picking one from the possible themes
+(define possible-themes '("ayu_mirage" "tokyonight_storm" "catppuccin_macchiato"))
+
+(define (select-random lst)
+  (let ([index (rand::rng->gen-range rng 0 (length lst))]) (list-ref lst index)))
```

Comment:
The steel random lib has changed, so this demo script needs this change to work:

```suggestion
;; Picking one from the possible themes
(define possible-themes '("ayu_mirage" "tokyonight_storm" "catppuccin_macchiato"))

(define (select-random lst)
  (let ([index (rand::rng->gen-range 0 (length lst))]) (list-ref lst index)))
```